### PR TITLE
Add audio support and simplify the pipeline

### PR DESCRIPTION
**Pull request by wkozyra95**
Main changes:
- add audio support
- remove the need to send a register_output request
- refactor that removes most of the state

Known issue:
- if output is unregistered to early it might crash because write to TCP socket is failing

### CODE REVIEW

**Review by WojciechBarczynski — APPROVED** [review]
LGTM

File: lib/membrane/video_compositor/video_compositor.ex [hunk 1]
```diff
@@ -232,29 +249,86 @@ defmodule Membrane.LiveCompositor do
                 Defines how the LiveCompositor bin should start-up a LiveCompositor server.
 
                 Available options:
-                - :start_locally - LC server is automatically started.
-                - :already_started - LiveCompositor bin assumes, that LC server is already started and is available on a specified port.
-                When this option is selected, the `api_port` option need to specify an exact port number (not a range).
+                - `:start_locally` - LC server is automatically started.
+                - `{:start_locally, path}` - LC server is automatically started, but different binary
```
[the remark this answers is not in the document]
Comment (wkozyra95) [reply lib/membrane/video_compositor/video_compositor.ex:253]: no, it's path to the binary

File: lib/membrane/video_compositor/streams_handler.ex [hunk 2]
```diff
@@ -1,22 +1,135 @@
...
+            offset -> Membrane.Time.as_milliseconds(offset, :round)
+          end
+      }
+      |> Request.send_request(state.lc_port)
+
+    case response do
+      {:ok, response} -> {:ok, response.body["port"]}
+      {:error_response_code, err} -> {:error, err}
+      {:error, err} -> {:error, err}
+    end
```
[the remark this answers is not in the document]
Comment (wkozyra95) [reply lib/membrane/video_compositor/streams_handler.ex:38]: I don't think this is more readable

[the remark this answers is not in the document]
Comment (WojciechBarczynski) [reply lib/membrane/video_compositor/streams_handler.ex:38]: BTW Does this function return `:inet.port_number()`? Isn't `response.body["port"]` a string?

[the remark this answers is not in the document]
Comment (WojciechBarczynski) [reply lib/membrane/video_compositor/streams_handler.ex:38]: This repeats in every function in this module, so I would extract this to some helper function and add spec to it

[the remark this answers is not in the document]
Comment (wkozyra95) [reply lib/membrane/video_compositor/streams_handler.ex:38]: > Isn't response.body["port"] a string?

no, it's a number https://github.com/membraneframework/video_compositor/blob/master/src/api.rs#L73

> This repeats in every function in this module, so I would extract this to some helper function and add spec to it

You just restated the original sentence from the first comment. I understand that code is repeated and I understand that you are suggesting to move it to helper function. I still stand by my original statement.

Only way to remove this repeated code that improves readability would be to change signature of send_request so it returns errors that can be easier matched e.g. 

```
{:error, {:error_response, response} | {:network_error, err}}
```

**Pull request merged**